### PR TITLE
Changes done for ip config view.

### PR DESCRIPTION
**Pull request by Srichandra03**
1. Updated the slave update to not clearing existing slaves.
2. Updated the scl and sda pin combo boxes to contain disabled pins.
3. Removed disabled pins row.

### CODE REVIEW

**Comment by chiruaj** [comment]
1. Updating the IP pin configuration changes when the IP configuration changes.
2. Prepopulating the directory path for the repo, tool chain, and output location.

File: src/main/java/com/board/config/boardconfiggui/LoadDataController.java [hunk 1]
```diff
@@ -206,4 +223,32 @@ private void openBoardNameDialog() {
...
+     *                       </ul>
+     * @return The initial directory path corresponding to the source node type, or null if not found.
+     */
+    private String getInitialDirectoryPath(String sourceNodeType) {
+        if (StringUtils.isEmpty(sourceNodeType)) {
+            return null;
+        }
+
+        return switch (sourceNodeType) {
+            case "xmlBtn" -> xmlFolderPath;
```
Comment (Gopal-moyya) [inline src/main/java/com/board/config/boardconfiggui/LoadDataController.java]: Define values in constants files

File: src/main/java/com/board/config/boardconfiggui/ui/utils/IpPinConfigListCell.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.board.config.boardconfiggui.ui.utils;
+
+import com.board.config.boardconfiggui.data.IpPinConfig;
+
+public class IpPinConfigListCell extends javafx.scene.control.ListCell<IpPinConfig> {
+  @Override
+  protected void updateItem(IpPinConfig item, boolean empty) {
+    super.updateItem(item, empty);
+    if (item == null || empty) {
+      setText("select");
```
Comment (Gopal-moyya) [inline src/main/java/com/board/config/boardconfiggui/ui/utils/IpPinConfigListCell.java]: select is already available in the constants

File: src/main/java/com/board/config/boardconfiggui/data/outputmodels/pinconfig/PinConfig.java [hunk 3]
```diff
@@ -122,24 +122,24 @@ public void removePinConfig(String portNumber, String pinNumber) {
...
 
     /**
-     * Method to get all the bypass enabled pin configurations of the current port.
+     * Method to get all the bypass disabled pin configurations of the current port.
      *
-     * @return bypass supported pin configurations
+     * @return bypass disabled pin configurations
      */
-    public Map<String, List<String>> getBypassConfiguredPins() {
+    public Map<String, List<String>> getBypassDisabledPins() {
```
Comment (Gopal-moyya) [inline src/main/java/com/board/config/boardconfiggui/data/outputmodels/pinconfig/PinConfig.java:129]: You have changed the method name however didn't observe any changes related to the calling function in the commit. can you check once

File: src/main/java/com/board/config/boardconfiggui/IpConfigController.java [hunk 4]
```diff
@@ -66,24 +67,33 @@ public class IpConfigController implements Initializable, BoardPageDataSaverInte
...
   }
 
   @FXML
   private void onSDAPinUpdate() {
-    String sdaPin = sdaChoiceBox.getValue();
-    Pair<String, String> sdaPinPair = getPinPortPair(sdaPin);
-    ipConfigModel.setSdaPort(sdaPinPair.getKey());
-    ipConfigModel.setSdaPin(sdaPinPair.getValue());
+    IpPinConfig ipPinConfig = sdaChoiceBox.getValue();
+    if (Objects.isNull(ipPinConfig)) {
```
Comment (Gopal-moyya) [inline src/main/java/com/board/config/boardconfiggui/IpConfigController.java:92]: Don't we need to reset when the user chooses the option as "select" from the list?

File: src/main/java/com/board/config/boardconfiggui/IpConfigController.java [hunk 5]
```diff
@@ -429,4 +422,59 @@ private void removeFromRepo() {
...
+   */
+  private void prepareIpPinConfigData() {
+    // Get the SCL port and PIN from the IP configuration model
+    String sclPortName = ipConfigModel.getSclPort();
+    String sclPinName = ipConfigModel.getSclPin();
+
+    if (StringUtils.isNotEmpty(sclPortName) && StringUtils.isNotEmpty(sclPinName)) {
+
+      if (Objects.nonNull(sclIpConfigModel) &&
+              !StringUtils.equals(sclIpConfigModel.getPinName(), sclPinName)) {
```
Comment (Gopal-moyya) [inline src/main/java/com/board/config/boardconfiggui/IpConfigController.java]: Also check for the portName

File: src/main/java/com/board/config/boardconfiggui/IpConfigController.java [hunk 6]
```diff
@@ -429,4 +422,59 @@ private void removeFromRepo() {
...
+      // If no existing configuration, create and save a new one
+      prepareAndSaveIpPinConfig(sclPortName, sclPinName);
+    }
+
+    // Get the SDA port and PIN from the IP configuration model
+    String sdaPortName = ipConfigModel.getSdaPort();
+    String sdaPinName = ipConfigModel.getSdaPin();
+    if (StringUtils.isNotEmpty(sdaPortName) && StringUtils.isNotEmpty(sdaPinName)) {
+
+      if (Objects.nonNull(sdaIpConfigModel) &&
```
Comment (Gopal-moyya) [inline src/main/java/com/board/config/boardconfiggui/IpConfigController.java:460]: Also check for the port name

**Review by Gopal-moyya — APPROVED** [review]

File: src/main/java/com/board/config/boardconfiggui/ui/utils/IpPinConfigListCell.java (hunk 2 above)
Comment (Srichandra03) [reply src/main/java/com/board/config/boardconfiggui/ui/utils/IpPinConfigListCell.java]: Updated to Constants.

File: src/main/java/com/board/config/boardconfiggui/IpConfigController.java (hunk 4 above)
Comment (Srichandra03) [reply src/main/java/com/board/config/boardconfiggui/IpConfigController.java:92]: We are not allowing user to select the default option as it is mandatory field.

File: src/main/java/com/board/config/boardconfiggui/IpConfigController.java (hunk 5 above)
Comment (Srichandra03) [reply src/main/java/com/board/config/boardconfiggui/IpConfigController.java]: Updated

File: src/main/java/com/board/config/boardconfiggui/IpConfigController.java (hunk 6 above)
Comment (Srichandra03) [reply src/main/java/com/board/config/boardconfiggui/IpConfigController.java:460]: Updated

**Pull request merged**